src/vacation_calendar/chore.py

```diff
@@ -0,0 +1,185 @@
+        ttype (TimeOffType): the type of time off; ROL for PaidTimeOff, VAC for Vacation.
+    Returns:
+        (pd.DataFrame): the df with the added time off
+    Raises:
+        ValueError if:
+        - `value` is higher than the `WORKING_HOURS`
+        - the chosen date is not present in the calendar
+        - the chosen date is a weekend or a holiday
+    """
+    if value > WORKING_HOURS:
+        raise ValueError("The amount per day cannot be higher than the working hours.")
+
+    mask = df.DATE == date
+
+    if df[mask].empty:
+        raise ValueError(f"The chosen date '{date}' does not exist in the calendar")
+
+    if (df[mask][["IS_WEEKEND", "IS_HOLIDAY"]]).any().any():
+        raise ValueError("The chosen day is a weekend or holiday")
+
+    df.loc[mask, ttype] = value
+    return df
+
+
+@add_to_calendar.register
+def add_range_to_calendar(
+    date: list, df: pd.DataFrame, value: float, ttype: TimeOffType
+) -> pd.DataFrame:
+    """
+    Add the input value to the calendar, in the passed date range. If weekends or holidays fall in the range,
+    they will be removed.
+    Args:
+        date (Union[List[datetime.date], List[str]]): list containing the start and the end dates in which
+            the time off will be taken
+        df (pd.DataFrame): input calendar df
+        value (float): amount of time off hours. Canno be higher than the `WORKING_HOURS`
+        ttype (TimeOffType): the type of time off; ROL for PaidTimeOff, VAC for Vacation.
+    Returns:
+        (pd.DataFrame): the df with the added time off
+    Raises:
+        ValueError if:
+        - the start date is after the end date
+        - `value` is higher than the `WORKING_HOURS`
+        - the chosen dates is not present in the calendar
+    """
+    start_date = date[0]
+    end_date = date[1]
+    if start_date > end_date:
+        raise ValueError(
+            f"The start date cannot be after the end date. Start: f{start_date}, End: {end_date}"
+        )
+
+    if value > WORKING_HOURS:
+        raise ValueError("The amount per day cannot be higher than the working hours.")
+
+    mask = (df.DATE >= start_date) & (df.DATE <= end_date)
+    mask &= ~df[mask][["IS_HOLIDAY", "IS_WEEKEND"]].any(axis=1)
+    if df[mask].empty:
+        raise ValueError(f"The chosen dates range'{date}' does not exist in the calendar")
+
+    df.loc[mask, ttype] = value
+    return df
+
+
+
+def _compute_residual_time(df: pd.DataFrame, time_type: str) -> pd.DataFrame:
+    """
+    Given the monthly aggregated dataframe, compute the residual time off/vacation
+    using the time off of the previous year and the time generated during the whole year.
+    Returns the aggregated dataframe with new columns
+    Args:
+        df (pd.DataFrame): the input dataframe
+        time_type (str): the type of time off. It can be ROL for paid time off or VAC
+            for vacation.
+    """
+    MAT = getattr(TimeOffType, f"{time_type}_MAT")
```

Comment:
What's the difference between `MAT = getattr(TimeOffType, f"{time_type}_MAT")` and `MAT = 
TimeOffType[f"{time_type}_MAT"]` if there is one?
I always used the direct __getitem__ call of the Enum, both of the approaches raise an Attribute or Key Error if I'm not wrong

Reply:
mmm if I'm not wrong no difference at all!